### ISSUE TITLE
REST API problems v2

### ISSUE THREAD

**Issue by BOHEUS** [twentyhq/twenty#7657]
@charlesBochet @martmull I've found some issues with REST API again, this time with endpoints in Metadata group, like earlier, 1 comment per issue to keep consistency

**Comment by charlesBochet** [comment]
Thank you! We are starting covering REST APIs with integration tests. We will be protected soon

**Comment by BOHEUS** [comment]
While creating new objects, user must give `labelIdentifierFieldMetadataId` and `imageIdentifierFieldMetadataId` but where these ids can be found in first place?
![image](https://github.com/user-attachments/assets/9ffa4da7-17f4-4fcb-a58e-dd0ea51a2186)

**Comment by BOHEUS** [comment]
I can update view sort and make it inactive which results in app's crash, I don't think that end user should be able to update all objects in first place but only those visible in Data model tab in Settings
![image](https://github.com/user-attachments/assets/2623ae6c-3f0b-4763-9885-71985e2b88a7)

**Comment by BOHEUS** [comment]
Even funnier (or scarier) fact: I can delete view sort which makes app non-usable so there's another problem
![image](https://github.com/user-attachments/assets/b2a98dbb-594d-4d94-87d0-382bd367067b)

**Comment by FelixMalfait** [comment]
Good feedback thanks @BOHEUS!

**Comment by BOHEUS** [comment]
Thanks but who said I'm done? 😂

**Comment by BOHEUS** [comment]
If there's 400 error, `Can't` word is missing apostrophe 
![image](https://github.com/user-attachments/assets/bbfe38da-733a-48d3-b0c8-c08c7b78f618)

**Comment by BOHEUS** [comment]
Using [this endpoint](https://twenty.com/developers/rest-api/metadata#/operations/deleteOneRelation) I'm able to delete relation between standard objects which normally can't happen
![image](https://github.com/user-attachments/assets/9da643f2-3f9a-4b22-96bc-e4dc948f208d)
![image](https://github.com/user-attachments/assets/512e4bbd-0e75-473d-8cf4-494272a55429)

**Comment by BOHEUS** [comment]
Using [this endpoint](https://twenty.com/developers/rest-api/metadata#/operations/createOneField) I'm able to add fields to objects which shouldn't have more fields like view

**Comment by Vanshika-Dargan** [comment]
I am willing to pick this up.
/assign

**Comment by BOHEUS** [comment]
Another day, another batch of bugs, this time from core

If user creates a new API key using API, then API doesn't return the API key which is totally pointless as API key is needed for connection (also, it seems like API keys created via API aren't visible in UI for some reason yet they exist in database?)
![image](https://github.com/user-attachments/assets/1344d22a-a722-40ee-8106-ef3371f2e0d0)

**Comment by BOHEUS** [comment]
I don't really understand attachments:
- what are possible types of attachments?
- if we want to upload a new attachment (which I understand is a file), why API requires path to it but not the file itself?
- why fullPath of attachment contains double token parameter?
- why am I able to overwrite the file extension/format of an attachment?
- if I overwrite a type of the attachment, attachment can't be downloaded
![image](https://github.com/user-attachments/assets/827aa24c-5a2a-496b-9fce-637245329c1d)

**Comment by BOHEUS** [comment]
Blocklists API aren't working properly, creating new or updating existing blocklists throw error about missing user id (which is weird since request workspaceMemberId is provided)
![image](https://github.com/user-attachments/assets/4bb7fda7-81f4-4ddb-8f7e-f8c3dd8f3879)

**Comment by BOHEUS** [comment]
Another bug: finding messages (aka emails) is impossible as it requires messageThreadId in filter but despite providing it like this `messageThreadId[eq]:\"20202020-634a-4fde-aa7c-28a0eaf203ca\"`, API still returns 400 error code
![Image](https://github.com/user-attachments/assets/03409e4d-0f63-439e-a238-8876c1834bbc)

**Comment by charlesBochet** [comment]
@bosiraphael FYI :)

**Comment by BOHEUS** [comment]
@charlesBochet @martmull new bug, this time reported by one of users, it's not possible to filter records by values from select or multi-select fields via API (but it's possible in UI) https://discord.com/channels/1130383047699738754/1324439805337141340

![Image](https://github.com/user-attachments/assets/cd130ff0-f8f0-40b9-a992-cad330443ea2)

**Comment by BOHEUS** [comment]
Another bug, if you create the same object twice (but with different icon), then error is thrown in UI and in logs, additionally whole app is unusable
![Image](https://github.com/user-attachments/assets/d0f83910-9c63-4f6f-bec3-9114c983f46b)

**Logs:**
```
/home/user/Desktop/projects/twenty_workspace/twenty/packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts:531
      throw new ObjectMetadataException(
            ^
Error: Object already exists
    at ObjectMetadataService.validatesNoOtherObjectWithSameNameExistsOrThrows (/home/user/Desktop/projects/twenty_workspace/twenty/packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts:531:13)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by FelixMalfait** [comment]
@martmull are you sure we can close this?

**Comment by martmull** [comment]
> Another day, another batch of bugs, this time from core
> 
> If user creates a new API key using API, then API doesn't return the API key which is totally pointless as API key is needed for connection (also, it seems like API keys created via API aren't visible in UI for some reason yet they exist in database?) ![image](https://github.com/user-attachments/assets/1344d22a-a722-40ee-8106-ef3371f2e0d0)

Yes, in fact you need to call the graphql `generateApiKeyToken` endpoint to generate the api key Bearer token. No token are stored in db. The generated tokens contains the apiKeyId information so we can revoke them
The rest api on api keys is a little bit useless indeed. The only useful thing is revoke apiKey via rest api

**Comment by martmull** [comment]
> I don't really understand attachments:
> 
> * what are possible types of attachments?
> * if we want to upload a new attachment (which I understand is a file), why API requires path to it but not the file itself?
> * why fullPath of attachment contains double token parameter?
> * why am I able to overwrite the file extension/format of an attachment?
> * if I overwrite a type of the attachment, attachment can't be downloaded
>   ![image](https://github.com/user-attachments/assets/827aa24c-5a2a-496b-9fce-637245329c1d)

Indeed, we may remove create and update attachment verbs from api key, what do you think @charlesBochet @FelixMalfait?

**Comment by martmull** [comment]
> I can update view sort and make it inactive which results in app's crash, I don't think that end user should be able to update all objects in first place but only those visible in Data model tab in Settings ![image](https://github.com/user-attachments/assets/2623ae6c-3f0b-4763-9885-71985e2b88a7)

Yes, those standard objects (not visible in settings) might be removed from the rest api to avoid app crash what do you think @FelixMalfait @charlesBochet ?

**Comment by martmull** [comment]
> Even funnier (or scarier) fact: I can delete view sort which makes app non-usable so there's another problem ![image](https://github.com/user-attachments/assets/b2a98dbb-594d-4d94-87d0-382bd367067b)

same

**Comment by martmull** [comment]
> Blocklists API aren't working properly, creating new or updating existing blocklists throw error about missing user id (which is weird since request workspaceMemberId is provided) ![image](https://github.com/user-attachments/assets/4bb7fda7-81f4-4ddb-8f7e-f8c3dd8f3879)

can't reproduce, worked well with workspaceMemeberId

<img width="1238" alt="Image" src="https://github.com/user-attachments/assets/9cbfa24a-a59d-4b87-984b-76b9301063be" />

**Comment by martmull** [comment]
> While creating new objects, user must give `labelIdentifierFieldMetadataId` and `imageIdentifierFieldMetadataId` but where these ids can be found in first place? ![image](https://github.com/user-attachments/assets/9ffa4da7-17f4-4fcb-a58e-dd0ea51a2186)

Must not, those fields are optional

**Comment by FelixMalfait** [comment]
Answered here: https://github.com/twentyhq/twenty/issues/10776
